bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -203,27 +270,17 @@ private void createJmDNSByAddress(InetAddress address) {
 
     @Override
     public void onChanged(List<CidrAddress> added, List<CidrAddress> removed) {
-        // remove jmdns instances that no longer exist due to interface changed
-        for (CidrAddress cidrAddress : removed) {
-            InetAddress inetAddr = cidrAddress.getAddress();
-            if (jmdnsInstances.containsKey(inetAddr)) {
-                JmDNS jmdns = jmdnsInstances.get(inetAddr);
-                closeQuietly(jmdns);
-                jmdnsInstances.remove(inetAddr);
-                logger.debug("mDNS service has been removed ({} for IP {})", jmdns.getName(),
-                        inetAddr.getHostAddress());
-            }
-        }
+        logger.debug("ip address change: added {}, removed {}", added, removed);
+        deactivate();
```

Comment:
Implementing delta changes is considerable complex with the new options, because this method also has to deal with configuration changes for the new options. Restarting the listeners takes less than a second, so I would like avoid this effort.

Reply:
Ok, understood and agreed.